modules/consensus/block_validation.go

```diff
@@ -63,7 +64,7 @@ func checkTarget(b types.Block, target types.Target) bool {
 // ValidateBlock validates a block against a minimum timestamp, a block target,
 // and a block height. Returns nil if the block is valid and an appropriate
 // error otherwise.
-func (bv stdBlockValidator) ValidateBlock(b types.Block, minTimestamp types.Timestamp, target types.Target, height types.BlockHeight) error {
+func (bv stdBlockValidator) ValidateBlock(b types.Block, minTimestamp types.Timestamp, target types.Target, height types.BlockHeight, log *persist.Logger) error {
```

Comment:
preliminary testing shows that an `EncodedLen` function would only be slightly faster:
```
BenchmarkLen-4          1000000   1465 ns/op   54.58 MB/s   472 B/op   27 allocs/op
BenchmarkLenMarshal-4 	1000000   1798 ns/op   44.48 MB/s   712 B/op   28 allocs/op
```

The quickest way to speed this up would be to add an `EncodedLen` to the `Block` type directly, which would in turn depend on an `EncodedLen` method of the `Transaction` type. These could conceivably bypass a lot of the encoding and therefore run much faster.

My guess is that `len(encoding.Marshal(x))` isn't much of a bottleneck though.